### PR TITLE
gh-2936 Relay service integration

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #2936

Changes proposed in this pull request:
- Gelato relay service
- Remaining relays request
- Relay request

### CODE REVIEW

File: Multisig/Cross-layer/Configuration/Config.Example.xcconfig [hunk 1]
```diff
@@ -50,6 +50,11 @@ _CLAIMING_DATA_URL_DEV        = https:/$()/safe-claiming-app-data.staging.5afe.d
 _CLAIMING_DATA_URL_STAGING    = https:/$()/safe-claiming-app-data.staging.5afe.dev/
 _CLAIMING_DATA_URL_PROD       = https:/$()/safe-claiming-app-data.gnosis-safe.io/
 
+// Gelato relay service
+_RELAY_URL_DEV                = https:/$()/
```
Comment (biafra23) [inline Multisig/Cross-layer/Configuration/Config.Example.xcconfig:54]: Can you provide an endpoint URL for testing? Or do you use an instance running on your laptop?

Comment (elgatovital) [reply Multisig/Cross-layer/Configuration/Config.Example.xcconfig:54]: I will adjust URLs after service is deployed.

File: Multisig/Data/Services/Safe Relay Service/SafeGelatoRelayService.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    func execute<T: JSONRequest>(request: T) throws -> T.ResponseType {
+        try httpClient.execute(request: request)
+    }
+
+    func asyncExecute<T: JSONRequest>(request: T, completion: @escaping (Result<T.ResponseType, Error>) -> Void) -> URLSessionTask? {
+        httpClient.asyncExecute(request: request, completion: completion)
+    }
+
+    // Returns mocked response in completion
+    func asyncExecuteMock<T: JSONRequest>(request: T, completion: @escaping (Result<T.ResponseType, Error>) -> Void) -> URLSessionTask? {
```
Comment (biafra23) [inline Multisig/Data/Services/Safe Relay Service/SafeGelatoRelayService.swift:39]: Why do we need this?

Comment (biafra23) [reply Multisig/Data/Services/Safe Relay Service/SafeGelatoRelayService.swift:39]: Can we use it to have a unit test that returns a predefined response? For example to test that the number in `expiresAt` is converted to a Date?

**Review by biafra23 — APPROVED** [review]
From what I can tell without running it, it looks good.

File: Multisig/Data/Services/Safe Relay Service/SafeGelatoRelayService.swift (hunk 2 above)
Comment (elgatovital) [reply Multisig/Data/Services/Safe Relay Service/SafeGelatoRelayService.swift:39]: this was the idea behind it. Unfortunately we are not using MockJSONHttpClient in our tests yet.

**Pull request merged**